client/ayon_core/hosts/substancepainter/plugins/load/load_mesh.py

```diff
@@ -8,10 +10,133 @@
+            "template_options": QtWidgets.QComboBox(),
+            "import_cameras": QtWidgets.QCheckBox("Import Cameras"),
+            "preserve_strokes": QtWidgets.QCheckBox("Preserve Strokes"),
+            "clickbox": QtWidgets.QWidget(),
+            "combobox": QtWidgets.QWidget(),
+            "buttons": QtWidgets.QDialogButtonBox(
+                QtWidgets.QDialogButtonBox.Ok
+                | QtWidgets.QDialogButtonBox.Cancel)
+        }
+
+        self.widgets["template_options"].addItems(self.template_names)
+
+        template_name = self.widgets["template_options"].currentText()
+        self._update_to_match_template(template_name)
+        # Build clickboxes
+        layout = QtWidgets.QHBoxLayout(self.widgets["clickbox"])
+        layout.addWidget(self.widgets["import_cameras"])
+        layout.addWidget(self.widgets["preserve_strokes"])
+        # Build combobox
+        layout = QtWidgets.QHBoxLayout(self.widgets["combobox"])
+        layout.addWidget(self.widgets["template_options"])
+        # Build buttons
+        layout = QtWidgets.QHBoxLayout(self.widgets["buttons"])
+        # Build layout.
+        layout = QtWidgets.QVBoxLayout(self)
+        layout.addWidget(self.widgets["label"])
+        layout.addWidget(self.widgets["combobox"])
+        layout.addWidget(self.widgets["clickbox"])
+        layout.addWidget(self.widgets["buttons"])
+
+        self.widgets["template_options"].currentTextChanged.connect(
+            self._update_to_match_template)
+        self.widgets["buttons"].accepted.connect(self.on_accept)
+        self.widgets["buttons"].rejected.connect(self.on_reject)
+
+    def on_accept(self):
+        self.configuration = self.get_project_configuration()
+        self.close()
+
+    def on_reject(self):
+        self.close()
+
+    def _update_to_match_template(self, template_name):
+        template = get_template_by_name(template_name, self.project_templates)
+        self.widgets["import_cameras"].setChecked(template["import_cameras"])
+        self.widgets["preserve_strokes"].setChecked(
+            template["preserve_strokes"])
+
+    def get_project_configuration(self):
+        templates = self.project_templates
+        template_name = self.widgets["template_options"].currentText()
+        template = get_template_by_name(template_name, templates)
+        template = copy.deepcopy(template) # do not edit the original
+        template["import_cameras"] = self.widgets["import_cameras"].isChecked()
+        template["preserve_strokes"] = (
+            self.widgets["preserve_strokes"].isChecked()
+        )
+        for key in ["normal_map_format",
+                    "project_workflow",
+                    "tangent_space_mode"]:
+            template[key] = _convert(template[key])
+        return template
+
+    @classmethod
+    def prompt(cls, templates):
+        dialog = cls(templates)
+        dialog.exec_()
+        dialog.deleteLater()
+
+        return dialog.configuration
```

Comment:
> Just to be safe it may be better to get the attribute from the object before marking it for deletion? It should be relatively safe from the python side as it currently is - but I'm not sure what the C++ side of things of Qt might be doing there.
> What do you think?

See [comment](https://github.com/ynput/ayon-core/pull/400#discussion_r1572277502).

```suggestion
        configuration = dialog.configuration
        dialog.deleteLater()
        return configuration
```